src/core/virtualize.ts

```diff
@@ -1,6 +1,66 @@
+import { IGridConfig, IGridScrollPosition } from "../interface";
+
 /**
  * Virtualize core.
  */
 export class Virtualize {
-    
+
+    /**
+     * Available grid height of virtualize.
+     */
+    private availableGridHeight: number;
+
+    /**
+     * Data set of virtualize.
+     */
+    private dataSet: any[];
+
+    /**
+     * Gets rows per index.
+     */
+    private get rowsPerIndex(): number {
+        return this.availableGridHeight / this.gridConfig.rowHeight;
+    }
+
+    /**
+     * Creates an instance of virtualize.
+     * @param gridConfig grid configuration.
+     * @param shadowRoot shadow root.
+     */
+    constructor(private gridConfig: IGridConfig, private shadowRoot: ShadowRoot) {
+        this.availableGridHeight = 200;
```

Comment:
```suggestion
        this.availableGridHeight = 200;
```
This shall be provided by a dynamic height calculator in future.